flow-server/src/main/java/com/vaadin/flow/router/QueryParameters.java

```diff
@@ -240,6 +242,52 @@ public String getQueryString() {
                 .collect(Collectors.joining(PARAMETERS_SEPARATOR));
     }
 
+    /**
+     * Return new QueryParameters excluding given parameters by names.
+     *
+     * @param keys
+     *            Names of the parameters to be excluded
+     * @return QueryParameters
+     */
+    public QueryParameters excluding(String... keys) {
```

Comment:
I wonder if it could be better to have at least one mandatory key

```suggestion
    public QueryParameters excluding(String key, String... otherKeys) {
```

Or short-circuiting to `new QueryParameters(parameters)` or `this` if the keys array is empty

Reply:
Done